src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -114,10 +112,25 @@ private CloudMessageSpool setupSpooler() {
      * @throws ServiceLoadException thrown if the service cannot be located
      */
     private CloudMessageSpool getPersistenceSpoolGGService()
-            throws ServiceLoadException {
+            throws ServiceLoadException, IOException {
         GreengrassService locatedService = kernel.locate(config.getPersistenceSpoolServiceName());
         if (locatedService instanceof CloudMessageSpool) {
-            return (CloudMessageSpool) locatedService;
+            CloudMessageSpool persistenceSpool = (CloudMessageSpool) locatedService;
+            persistenceSpool.initializeSpooler();
```

Comment:
initialize needs to be able to throw and you need to handle that case when it does throw.

Reply:
It will be covered under `IOException` above.